napari/utils/settings/_manager.py

```diff
@@ -0,0 +1,133 @@
+"""Settings management.
+"""
+
+import os
+from json import dumps
+from pathlib import Path
+
+import toml
+from pydantic import ValidationError
+from yaml import safe_dump, safe_load
+
+from napari.utils.settings._defaults import (
+    ApplicationSettings,
+    ConsoleSettings,
+)
+
+
+class SettingsManager:
+    """"""
+
+    def __init__(self, config_path=None):
+        self._config_path = Path.home() / ".napari" / "settings"
+        self._settings = {}
+        self._defaults = {}
+        self._models = {}
+        self._plugins = []
+        self._format = "yaml"
+
+        if not self._config_path.is_dir():
+            os.makedirs(self._config_path)
+
+        self._load()
+
+    def __getattr__(self, attr):
+        if attr in self._settings:
+            return self._settings[attr]
+
+    def _get_file_name(self, settings):
+        """"""
+        key = settings.Config.title.replace(" ", "_").lower()
+        if key.endswith("_settings"):
+            key = key.replace("_settings", "")
+
+        return key
+
+    def _save(self):
+        """Save configuration to disk."""
+        data = {}
+        for key, model in self._settings.items():
+            data[key] = model.dict()
+
+        path = self._config_path / f"settings.{self._format}"
+        with open(path, "w") as fh:
+            fh.write(safe_dump(data))
```

Comment:
any concern here about overwriting the existing file? not sure if there is a scenario where there is unsaved changes in this file and is erased because of overwrite

Reply:
Hi @ziyangczi so currently we save on every update, so overwrites should not fall under that.

We are not checking for file changes (we could!)

Also I might need to look into https://pypi.org/project/atomicwrites/ to make sure saves are being atomic.